app/models/status.rb

```diff
@@ -403,29 +403,34 @@ def status_stat
   # Otherwise, the code is kept as close as possible to ActiveRecord::Persistence
   # code, and actually calls it if we are not handling a reblog.
   def self._insert_record(values)
-    return super unless values.is_a?(Hash) && values['reblog_of_id'].present?
+    return super unless values.is_a?(Hash) && values['reblog_of_id']&.value.present?
 
     primary_key = self.primary_key
     primary_key_value = nil
 
-    if primary_key
-      primary_key_value = values[primary_key]
-
-      if !primary_key_value && prefetch_primary_key?
+    if prefetch_primary_key? && primary_key
+      values[primary_key] ||= begin
         primary_key_value = next_sequence_value
-        values[primary_key] = primary_key_value
+        _default_attributes[primary_key].with_cast_value(primary_key_value)
       end
     end
 
-    # The following line is where we differ from stock ActiveRecord implementation
-    im = _compile_reblog_insert(values)
+    im = Arel::InsertManager.new
 
-    # Since we are using SELECT instead of VALUES, a non-error `nil` return is possible.
-    # For our purposes, it's equivalent to a foreign key constraint violation
-    result = connection.insert(im, "#{self} Create", primary_key || false, primary_key_value)
-    raise ActiveRecord::InvalidForeignKey, "(reblog_of_id)=(#{values['reblog_of_id']}) is not present in table \"statuses\"" if result.nil?
+    if values.empty?
+      im.insert(connection.empty_insert_statement_value(primary_key))
```

Comment:
I'd avoid this case to streamline the code, as we know for a fact `values` is not empty. Otherwise this looks good.